Lib/importlib/_bootstrap_external.py

```diff
@@ -221,280 +221,9 @@ def _write_atomic(path, data, mode=0o666):
-#     Python 3.12b1 3526 (Add instrumentation support)
-#     Python 3.12b1 3527 (Add LOAD_SUPER_ATTR)
-#     Python 3.12b1 3528 (Add LOAD_SUPER_ATTR_METHOD specialization)
-#     Python 3.12b1 3529 (Inline list/dict/set comprehensions)
-#     Python 3.12b1 3530 (Shrink the LOAD_SUPER_ATTR caches)
-#     Python 3.12b1 3531 (Add PEP 695 changes)
-#     Python 3.13a1 3550 (Plugin optimizer support)
-#     Python 3.13a1 3551 (Compact superinstructions)
-#     Python 3.13a1 3552 (Remove LOAD_FAST__LOAD_CONST and LOAD_CONST__LOAD_FAST)
-#     Python 3.13a1 3553 (Add SET_FUNCTION_ATTRIBUTE)
-#     Python 3.13a1 3554 (more efficient bytecodes for f-strings)
-#     Python 3.13a1 3555 (generate specialized opcodes metadata from bytecodes.c)
-#     Python 3.13a1 3556 (Convert LOAD_CLOSURE to a pseudo-op)
-#     Python 3.13a1 3557 (Make the conversion to boolean in jumps explicit)
-#     Python 3.13a1 3558 (Reorder the stack items for CALL)
-#     Python 3.13a1 3559 (Generate opcode IDs from bytecodes.c)
-#     Python 3.13a1 3560 (Add RESUME_CHECK instruction)
-#     Python 3.13a1 3561 (Add cache entry to branch instructions)
-#     Python 3.13a1 3562 (Assign opcode IDs for internal ops in separate range)
-#     Python 3.13a1 3563 (Add CALL_KW and remove KW_NAMES)
-#     Python 3.13a1 3564 (Removed oparg from YIELD_VALUE, changed oparg values of RESUME)
-#     Python 3.13a1 3565 (Oparg of YIELD_VALUE indicates whether it is in a yield-from)
-#     Python 3.13a1 3566 (Emit JUMP_NO_INTERRUPT instead of JUMP for non-loop no-lineno cases)
-#     Python 3.13a1 3567 (Reimplement line number propagation by the compiler)
-#     Python 3.13a1 3568 (Change semantics of END_FOR)
-#     Python 3.13a5 3569 (Specialize CONTAINS_OP)
-#     Python 3.13a6 3570 (Add __firstlineno__ class attribute)
-#     Python 3.14a1 3600 (Add LOAD_COMMON_CONSTANT)
-#     Python 3.14a1 3601 (Fix miscompilation of private names in generic classes)
-#     Python 3.14a1 3602 (Add LOAD_SPECIAL. Remove BEFORE_WITH and BEFORE_ASYNC_WITH)
-#     Python 3.14a1 3603 (Remove BUILD_CONST_KEY_MAP)
-
-#     Python 3.15 will start with 3650
-
-#     Please don't copy-paste the same pre-release tag for new entries above!!!
-#     You should always use the *upcoming* tag. For example, if 3.12a6 came out
-#     a week ago, I should put "Python 3.12a7" next to my new magic number.
-
-# MAGIC must change whenever the bytecode emitted by the compiler may no
-# longer be understood by older implementations of the eval loop (usually
-# due to the addition of new opcodes).
-#
-# Starting with Python 3.11, Python 3.n starts with magic number 2900+50n.
-#
-# Whenever MAGIC_NUMBER is changed, the ranges in the magic_values array
-# in PC/launcher.c must also be updated.
-
-MAGIC_NUMBER = (3603).to_bytes(2, 'little') + b'\r\n'
-
-_RAW_MAGIC_NUMBER = int.from_bytes(MAGIC_NUMBER, 'little')  # For import.c
+_RAW_MAGIC_NUMBER = int.from_bytes(MAGIC_NUMBER, 'little')  # for import.c
```

Comment:
This shouldn't be needed anymore.

Reply:
It's used in [a few places on the Python side](https://github.com/search?q=repo%3Apython%2Fcpython%20_RAW_MAGIC_NUMBER&type=code), and it's not the same value as `_imp._pyc_magic_number`, because it includes the `\r\n` in the lower bytes.  (But it is equalivalent to the result of `PyImport_GetMagicNumber()`).  

Not sure how best to deal with that -- we could add another member to `_imp` with that value, called `_pyc_magic_number_token` or something?

Reply:
`MAGIC_NUMBER` and `_RAW_MAGIC_NUMBER` are definitely coupled.  (There were added together in 77b2abd094998b89ff4d806b78683d4cce8d6d79.)  I'd say lets keep them together.

Also, there's no need for a leading underscore for either name when exposed by `_imp`.